ledger/participant-integration-api/src/main/scala/platform/localstore/PersistentIdentityProviderConfigStore.scala

```diff
@@ -0,0 +1,172 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.localstore
+
+import com.daml.ledger.api.domain
+import com.daml.ledger.api.domain.IdentityProviderId
+import com.daml.logging.{ContextualizedLogger, LoggingContext}
+import com.daml.metrics.{DatabaseMetrics, Metrics}
+import com.daml.platform.localstore.api.IdentityProviderConfigStore.{
+  IdentityProviderConfigExists,
+  IdentityProviderConfigNotFound,
+  IdentityProviderConfigWithIssuerExists,
+  Result,
+}
+import com.daml.platform.localstore.api.{IdentityProviderConfigStore, IdentityProviderConfigUpdate}
+import com.daml.platform.store.DbSupport
+
+import java.sql.Connection
+import scala.concurrent.{ExecutionContext, Future}
+
+class PersistentIdentityProviderConfigStore(
+    dbSupport: DbSupport,
+    metrics: Metrics,
+)(implicit executionContext: ExecutionContext)
+    extends IdentityProviderConfigStore {
+
+  private val backend = dbSupport.storageBackendFactory.createIdentityProviderConfigStorageBackend
+  private val dbDispatcher = dbSupport.dbDispatcher
+  private val logger = ContextualizedLogger.get(getClass)
+
+  override def createIdentityProviderConfig(identityProviderConfig: domain.IdentityProviderConfig)(
+      implicit loggingContext: LoggingContext
+  ): Future[Result[domain.IdentityProviderConfig]] = {
+    inTransaction(_.createIDPConfig) { implicit connection =>
```

Comment:
Consider using helper methods `wihtIdpConfig` and `withoutIdpConfig` similar to `withUser` and `withoutUser` in user management

Reply:
I actually considered and even had it, but in fact they're used once in this file, so there was no value to add it